cfgov/research_reports/jinja2/index.html

```diff
@@ -0,0 +1,114 @@
+{% set is_report = True %}
+{% extends 'layout-side-nav.html' %}
+
+{% import './pdf-download.html' as pdf_download with context %}
+{% import 'templates/streamfield-sidefoot.html' as streamfield_sidefoot with context %}
+
+{% block css %}
+{{ super() }}
+<link rel="stylesheet" href="{{ static('apps/research-reports/css/main.css') }}">
+{% endblock %}
+
+{% block content_main_modifiers -%}
+    {{ super() }} content__flush-bottom research-report
+{%- endblock %}
+
+{% if report_sections is not defined  %}
+    {% set report_sections = get_report_sections(request, page) %}
+{% endif %}
+
+{% if report_appendices is not defined  %}
+    {% set report_appendices = get_report_appendices(request, page) %}
+{% endif %}
+
+{% set researchers = get_researchers() %}
+
+
+{% macro report_part(sections, is_appendix) %}
+  {% for section in sections -%}
+    <h2 class="report-header">
+      <a
+        id="{{section.url[1:]}}"
+        href="{{section.url}}">
+          {{('Appendix ' if is_appendix else '') + section.numbering + section.title}}
+      </a>
+    </h2>
+
+    {{ section.body | safe }}
```

Comment:
Just to confirm: these need `| safe`  because they contain raw HTML that needs to go verbatim into the page?

Reply:
Yes. Without `| safe`, the html content would show up as html instead of as content.

Reply:
It would be nice if there was some kind of validation against improper HTML, for example stripping out `<script>` tags or the like. I realize that might add some extra complexity, though, depending on how careful you think we need to be about the input content.

Reply:
Interesting. What case are you hoping to protect against? I think the only ways that HTML could get into these models are (a) from our `parse_research_report` command, which is based on Pandoc, which wouldn't include a `<script>` tag, (b) entered manually by a content editor (who wouldn't have a reason to include a `<script>` tag), or (c) entered maliciously by someone unauthorized.  I felt okay using `| safe` because (a) is the intended use case and should be the overwhelming majority of where content comes from. 

Are you picturing a custom method to scan HTML and strip out undesired tags? And when are you picturing that happening--at the parsing step or on page load?

Reply:
Those are great questions. My comment came from a general feeling that putting fields behind `| safe` is generally something we should only do if we are really sure that it's safe to do so. In terms of case (b), there's also the possibility of something inadvertently getting included, either by Pandoc or by a content editor making a mistake. Maybe the risk is low enough -- after all, we already allow [RawHTMLBlocks](https://docs.wagtail.io/en/v2.9.3/topics/streamfield.html#rawhtmlblock) on pages, which does the same thing.

If we did want an extra layer of certainty about Pandoc output, either now or in future, we could do some post-processing of the soup, for example:

```py
# Given the soup generated when BS runs on the Pandoc output
>>> set(tag.name for tag in soup.find_all())
{'td', 'table', 'sup', 'u', 'thead', 'p', 'hr', 'h5', 'h2', 'ol', 'em', 'br', 'strong', 'tbody', 'th', 'section', 'h1', 'img', 'li', 'a', 'tr'}
# Check to make sure 'script' isn't in that list...
```

But as you mention, this wouldn't protect against tags entered in the editor unless this also happened on page save validation.

Reply:
Thanks for that idea. I think that can be a future enhancement, rather than going into this PR, so I'll add it to our backlog.